vm/actor/src/builtin/miner/mod.rs

```diff
@@ -1014,185 +1063,110 @@ impl Actor {
+                return Err(actor_error!(ErrInsufficientFunds, "insufficient funds {} for pre-commit deposit: {}", available_balance, total_deposit_required));
             }
-
-            state.add_pre_commit_deposit(&deposit_req).map_err(|e| {
-                actor_error!(
-                    ErrIllegalState,
-                    "failed to add pre-commit deposit {}: {}",
-                    deposit_req,
-                    e
-                )
-            })?;
-
-            let seal_proof = params.seal_proof;
-            let sector_number = params.sector_number;
-
-            state
-                .put_precommitted_sector(
-                    store,
-                    SectorPreCommitOnChainInfo {
-                        info: params,
-                        pre_commit_deposit: deposit_req,
-                        pre_commit_epoch: rt.curr_epoch(),
-                        deal_weight: deal_weight.deal_weight.clone(),
-                        verified_deal_weight: deal_weight.verified_deal_weight.clone(),
-                    },
-                )
-                .map_err(|e| {
-                    e.downcast_default(
-                        ExitCode::ErrIllegalState,
-                        format!("failed to write pre-committed sector {}", sector_number),
-                    )
-                })?;
-
-            // add precommit expiry to the queue
-            let max_seal_duration = max_prove_commit_duration(seal_proof).ok_or_else(|| {
-                actor_error!(
-                    ErrIllegalArgument,
-                    "no max seal duration set for proof type: {:?}",
-                    seal_proof
-                )
-            })?;
-
-            // The +1 here is critical for the batch verification of proofs. Without it, if a proof arrived exactly on the
-            // due epoch, ProveCommitSector would accept it, then the expiry event would remove it, and then
-            // ConfirmSectorProofsValid would fail to find it.
-            let expiry_bound = rt.curr_epoch() + max_seal_duration + 1;
-
-            state
-                .add_pre_commit_expiry(store, expiry_bound, sector_number)
+            state.add_pre_commit_deposit(&total_deposit_required)
+                .map_err(|e|
+                    actor_error!(
+                        ErrIllegalState,
+                        "failed to add pre-commit deposit {}: {}",
+                        total_deposit_required, e
+                ))?;
+            state.allocate_sector_numbers(store, &sector_numbers, CollisionPolicy::DenyCollisions)
+                .map_err(|e|
+                     e.wrap("failed to allocate sector numbers")
+                )?;
+            state.put_precommitted_sectors(store, chain_infos)
+                .map_err(|e|
+                    e.downcast_default(ExitCode::ErrIllegalState, "failed to write pre-committed sectors")
+                )?;
+            state.add_pre_commit_clean_ups(store, clean_up_events)
```

Comment:
`add_pre_commit_expirations`

Reply:
No, its changed to clean_ups intentionally https://github.com/filecoin-project/specs-actors/blob/master/actors/builtin/miner/miner_actor.go#L838